core/commands/pin.go

```diff
@@ -377,13 +378,18 @@ new pin and removing the old one.
 			return
 		}
 
-		fromc, err := core.ResolveToCid(req.Context(), n, from)
+		r := &path.Resolver{
```

Comment:
This construction is more and more frequent around, it might be good to store it in ipfs node struct or have helper for it.